mobile/src/modules/auth/components/details/TransactionListItem.tsx

```diff
@@ -0,0 +1,131 @@
+import { Transaction } from '@burstjs/core';
+import React from 'react';
+import { Image, TouchableOpacity, View } from 'react-native';
+import { transactionIcons } from '../../../../assets/icons';
+import { Text, TextAlign } from '../../../../core/components/base/Text';
+import { Colors } from '../../../../core/theme/colors';
+import { defaultSideOffset, FontSizes, Sizes } from '../../../../core/theme/sizes';
+import { getShortDateFromTimestamp } from '../../../../core/utils/date';
+import { NQTAmountToString } from '../../../../core/utils/numbers';
+
+interface Props {
+  transaction: Transaction;
+  accountRS: string;
+  onPress: (transaction: Transaction) => void;
+}
+
+const styles: any = {
+  view: {
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'nowrap',
+    paddingVertical: Sizes.MEDIUM
+  },
+  iconView: {
+    width: 40,
+    marginLeft: 0,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center'
+  },
+  mainView: {
+    marginRight: 0,
+    paddingRight: defaultSideOffset,
+    display: 'flex',
+    flexDirection: 'column',
+    flexWrap: 'wrap',
+    flexGrow: 1
+  },
+  hintView: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    marginBottom: Sizes.MEDIUM
+  },
+  dataView: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start'
+  },
+  icon: {
+    width: 20,
+    height: 20
+  },
+  incomingAmount: {
+    marginRight: Sizes.MEDIUM
+  },
+  outcomingAmount: {
+    marginLeft: Sizes.MEDIUM
+  }
+};
+
+export class TransactionListItem extends React.PureComponent<Props> {
+  isIncomingTransaction = () => {
+    const { accountRS, transaction } = this.props;
+    return accountRS === transaction.recipientRS;
+  }
+
+  handlePress = () => {
+    const { onPress, transaction } = this.props;
+    onPress(transaction);
+  }
+
+  renderIcon = () => {
+    const { confirmations = 0 } = this.props.transaction;
+
+    const icon = confirmations > 0
+      ? transactionIcons.done
+      : transactionIcons.waiting;
+
+    return (
+      <Image source={icon} style={styles.icon} />
+    );
+  }
+
+  render () {
+    const {
+      amountNQT = '0',
+      transaction = '',
+      timestamp = 0,
+      recipientRS = '',
+      senderRS = ''
+    } = this.props.transaction;
+    const isIncoming = this.isIncomingTransaction();
+    const amount = NQTAmountToString(amountNQT);
+    const accountRS = isIncoming ? senderRS : recipientRS;
+    const date = getShortDateFromTimestamp(timestamp);
+
+    return (
+      <TouchableOpacity style={styles.view} onPress={this.handlePress}>
```

Comment:
very nice! seems the icon in between the amount and the address is missing (it's a caret-looking icon, like >)

Reply:
There isn't enough space on iPhone SE :)